code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -198,7 +198,7 @@
 			X.visible_message(span_danger("\The [X] smashes \the [src]!"), \
 			span_danger("We smash \the [src]!"), null, 5)
 			take_damage(damage_amount, damage_type, damage_flag, effects, null, armor_penetration)
-		SEND_SIGNAL(X, COMSIG_XENOMORPH_ATTACK_CLOSET)
+		. = ..()
```

Comment:
actually I miss read return ..() was probably fine lolol.

But anyway, let's improve that proc. First you check for opened, and early return, and then another ealy return (checking intent harm and the bitfield), and then . = ..(), and then the rest of the proc

Reply:
imma do this tomorrow